src/components/virtual-gamepad-controls.js

```diff
@@ -0,0 +1,188 @@
+import nipplejs from "nipplejs";
+
+var THREE = AFRAME.THREE;
+var DEGREES = Math.PI / 180;
+var HALF_PI = Math.PI / 2;
+
+AFRAME.registerComponent("virtual-gamepad-controls", {
+  schema: {
+    movementSpeed: { default: 2 },
+    lookSpeed: { default: 60 }
+  },
+
+  init() {
+    // Setup gamepad elements
+    var leftTouchZone = document.createElement("div");
+    leftTouchZone.style.position = "absolute";
+    leftTouchZone.style.left = 0;
+    leftTouchZone.style.top = 0;
+    leftTouchZone.style.bottom = 0;
+    leftTouchZone.style.right = "50%";
+
+    var rightTouchZone = document.createElement("div");
+    rightTouchZone.style.position = "absolute";
+    rightTouchZone.style.left = "50%";
+    rightTouchZone.style.top = 0;
+    rightTouchZone.style.bottom = 0;
+    rightTouchZone.style.right = 0;
```

Comment:
I would put this in a CSS file and assign a className here instead.

Reply:
Actually, maybe this is fine since it's nice that the component is self-contained.

Reply:
We can use a webpack loader for this.